Sources/Subscription/Managers/SubscriptionManager.swift

```diff
@@ -104,70 +196,229 @@ public final class DefaultSubscriptionManager: SubscriptionManager {
         }
     }
 
-    // MARK: - Environment, ex SubscriptionPurchaseEnvironment
+    // MARK: - Environment
 
     @available(macOS 12.0, iOS 15.0, *) private func setupForAppStore() {
         Task {
             await storePurchaseManager().updateAvailableProducts()
         }
     }
 
-    // MARK: -
+    // MARK: - Subscription
 
-    public func loadInitialData() {
-        Task {
-            if let token = accountManager.accessToken {
-                _ = await subscriptionEndpointService.getSubscription(accessToken: token, cachePolicy: .reloadIgnoringLocalCacheData)
-                _ = await accountManager.fetchEntitlements(cachePolicy: .reloadIgnoringLocalCacheData)
-            }
-        }
-    }
-
-    public func refreshCachedSubscriptionAndEntitlements(completion: @escaping (_ isSubscriptionActive: Bool) -> Void) {
-        Task {
-            guard let token = accountManager.accessToken else { return }
+    public func loadInitialData() async {
 
-            var isSubscriptionActive = false
+        // Attempting V1 token migration
+        // IMPORTANT: This MUST be the first operation executed by Subscription
+        do {
+            if (try await oAuthClient.migrateV1Token()) != nil {
+                pixelHandler(.v1MigrationSuccessful)
 
-            defer {
-                completion(isSubscriptionActive)
+                // cleaning up old data
+                clearSubscriptionCache()
             }
+        } catch {
+            Logger.subscription.error("Failed to migrate V1 token: \(error, privacy: .public)")
```

Comment:
Shouldn't this error throw?  The VPN calls this at startup: what happens if the migration fails?

Reply:
If the migration fails there's nothing we can do, we just don't have a token and we stop and just send a pixel (added now in the vpn too)